services/ip-whitelist.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+var request = require('superagent');
+var errorMessages = require('../utils/error-messages');
+var logger = require('./logger');
+var ServiceUrlGetter = require('./service-url-getter');
+var ipUtil = require('ip-utils');
+var P = require('bluebird');
+var _ = require('lodash');
+
+var ipWhitelistRules = null;
+var useIpWhitelist = true;
```

Comment:
Very touchy/risky to set the value to true, if any unexpected error occurs. Especially for the majority of projects that wont use this feature and let any IP access the data